src/include/storage/store/column_chunk_flush.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include "storage/buffer_manager/bm_file_handle.h"
+#include "storage/compression/compression.h"
+#include "storage/compression/compression_float.h"
+#include "storage/store/column_chunk_metadata.h"
+
+namespace kuzu::storage {
+class CompressedFlushBuffer {
+    std::shared_ptr<CompressionAlg> alg;
+    const common::LogicalType& dataType;
+
+public:
+    CompressedFlushBuffer(std::shared_ptr<CompressionAlg> alg, const common::LogicalType& dataType)
+        : alg{std::move(alg)}, dataType{dataType} {}
+
+    CompressedFlushBuffer(const CompressedFlushBuffer& other) = default;
+
+    ColumnChunkMetadata operator()(const uint8_t* buffer, uint64_t bufferSize, BMFileHandle* dataFH,
+        common::page_idx_t startPageIdx, const ColumnChunkMetadata& metadata) const;
+};
+
+template<std::floating_point T>
+class CompressedFloatFlushBuffer {
```

Comment:
The problem is that the ALP exceptions are generated/handled in this function, not in `CompressionAlg` since they are stored in separate pages (while `CompressionAlg` only operates on a per-page level`). So if you take a look at the code, it is quite different and I don't think we can cleanly reuse `CompressedFlushBuffer` for this.
Same thing for `GetCompressionMetadata`.